### PR TITLE
Add point sprites

### PR DESCRIPTION
**Pull request by almarklein**
Not necessarily the ones useful for creating GUI elements. But more so for particle effects etc.

image

### CODE REVIEW

File: pygfx/materials/_points.py [hunk 1]
```diff
@@ -238,5 +238,34 @@ class PointsGaussianBlobMaterial(PointsMaterial):
...
+        self.sprite = sprite
+
+    @property
+    def sprite(self):
+        """The texture map specifying the sprite image.
+
+        The dimensionality of the map must be 2D. If None, it just shows a
+        uniform color.
+        """
+        return self._sprite
```
Comment (Korijn) [inline pygfx/materials/_points.py:262]: should this be stored on store?

File: pygfx/materials/_points.py [hunk 2]
```diff
@@ -238,5 +238,34 @@ class PointsGaussianBlobMaterial(PointsMaterial):
     """
 
 
+class PointsSpriteMaterial(PointsMaterial):
+    """A material to render points as sprite images.
+
+    Renders the provided texture at each point position. The images are square
+    and sized just like with a PointMaterial. The texture color is multiplierd
```
Comment (Korijn) [inline pygfx/materials/_points.py:245]:
```suggestion
    and sized just like with a PointMaterial. The texture color is multiplied
```

File: pygfx/materials/_points.py [hunk 3]
```diff
@@ -238,5 +238,34 @@ class PointsGaussianBlobMaterial(PointsMaterial):
...
+    with the point's "normal" color (as calculated depending on ``color_mode``).
+
+    The sprite texture is provided via ``.map``.
```
Comment (Korijn) [inline pygfx/materials/_points.py:248]: Isn't it `.sprite`?

File: pygfx/materials/_points.py (hunk 1 above)
Comment (almarklein) [reply pygfx/materials/_points.py:262]: Good point! I copied from `PointsMaterial.map`, but that one was faulty too :)

File: pygfx/materials/_points.py (hunk 3 above)
Comment (almarklein) [reply pygfx/materials/_points.py:248]: woops, yets  it is!

File: pygfx/materials/_points.py (hunk 1 above)
Comment (Korijn) [reply pygfx/materials/_points.py:262]: I think you missed `self._sprite` though?

Comment (almarklein) [reply pygfx/materials/_points.py]: VSCode refused to save the file because of conflicts with the typo-change :P

**Review by Korijn — APPROVED** [review]

**Pull request merged**